swarm/api/routes/db.py

```
@@ -141,6 +143,21 @@ async def rebuild_database(request: DBRebuildRequest):
+        for run_id in request.run_ids:
+            try:
+                validate_path_component(run_id, "run_id")
+            except ValueError as e:
+                raise HTTPException(
+                    status_code=400,
+                    detail={
+                        "error": "invalid_run_id",
+                        "message": str(e),
+                        "details": {"run_id": run_id},
+                    },
+                )
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)